libethcore/BlockInfo.cpp

```diff
@@ -145,64 +130,11 @@ void BlockInfo::populateFromHeader(RLP const& _header, Strictness _s, h256 const
 		throw;
 	}
 
-	if (number > ~(unsigned)0)
+	if (m_number > ~(unsigned)0)
```

Comment:
`m_number` is `u256` so perhaps `(u256)` would be more reasonable here. removing the cast may also be an option since i guess it'll be upcast to `u256` anyway (not sure if supported on all compilers though).